dev/com.ibm.ws.microprofile.faulttolerance.2.0/src/com/ibm/ws/microprofile/faulttolerance20/state/impl/RetryStateImpl.java

```diff
@@ -55,31 +55,31 @@ public RetryResult recordResult(MethodResult<?> methodResult) {
         if (methodResult.isFailure()) {
             // Failure case
             if (abortOn(methodResult.getFailure())) {
-                result.shouldRetry = false;
+                result.reason = RetryResultReason.EXCEPTION_IN_ABORTON;
             } else if (retryOn(methodResult.getFailure())) {
-                result.shouldRetry = true;
+                result.reason = RetryResultReason.EXCEPTION_IN_RETRYON;
             } else {
-                result.shouldRetry = false;
+                result.reason = RetryResultReason.EXCEPTION_NOT_IN_RETRYON;
             }
         } else {
             // Successful case
-            result.shouldRetry = false;
+            result.reason = RetryResultReason.NO_EXCEPTION;
         }
 
         // Capture whether this result was considered a failure by the Retry
-        boolean resultWasFailure = result.shouldRetry;
+        boolean resultWasFailure = result.shouldRetry();
```

Comment:
you're asking the result if it thinks it should retry and then inferring that the result was a failure. Perhaps the method should be `result.wasFailure()`?